compiler/rustc_metadata/src/creader.rs

```diff
@@ -306,7 +307,7 @@ impl CStore {
         deps
     }
 
-    fn crate_dependencies_in_reverse_postorder(&self, cnum: CrateNum) -> Vec<CrateNum> {
+    fn crate_dependencies_in_reverse_postorder(&self, cnum: CrateNum) -> IndexSet<CrateNum> {
         let mut deps = self.crate_dependencies_in_postorder(cnum);
         deps.reverse();
```

Comment:
This isn't related to your change, but for the one place that uses this reverse order, it would probably be better to take the normal order and just reverse the iterator, `.into_iter().rev()`.

https://github.com/rust-lang/rust/blob/13718eb788622ef8c998650451174570230d2971/compiler/rustc_metadata/src/creader.rs#L1229

This method could even return `impl Iterator` that's already reversed that way.